eel/src/lib.rs

```diff
@@ -494,34 +493,26 @@ impl LightningNode {
         error: PaymentError,
         payment_hash: &str,
     ) -> Result<()> {
+        self.data_store
+            .lock()
+            .unwrap()
+            .new_payment_state(payment_hash, PaymentState::Failed)?;
         match error {
-            PaymentError::Invoice(e) => {
-                self.data_store
-                    .lock()
-                    .unwrap()
-                    .new_payment_state(payment_hash, PaymentState::Failed)?;
-                Err(invalid_input(format!("Invalid invoice - {e}")))
-            }
-            PaymentError::Sending(e) => {
-                self.data_store
-                    .lock()
-                    .unwrap()
-                    .new_payment_state(payment_hash, PaymentState::Failed)?;
-                match e {
-                    RetryableSendFailure::PaymentExpired => Err(runtime_error(
-                        RuntimeErrorCode::SendFailure,
-                        format!("Failed to send payment - {e:?}"),
-                    )),
-                    RetryableSendFailure::RouteNotFound => Err(runtime_error(
-                        RuntimeErrorCode::NoRouteFound,
-                        "Failed to find a route",
-                    )),
-                    RetryableSendFailure::DuplicatePayment => Err(runtime_error(
-                        RuntimeErrorCode::SendFailure,
-                        format!("Failed to send payment - {e:?}"),
-                    )),
-                }
-            }
+            PaymentError::Invoice(e) => Err(invalid_input(format!("Invalid invoice - {e}"))),
+            PaymentError::Sending(e) => match e {
+                RetryableSendFailure::PaymentExpired => Err(runtime_error(
+                    RuntimeErrorCode::InvoiceExpired,
+                    "Invoice has expired",
+                )),
+                RetryableSendFailure::RouteNotFound => Err(runtime_error(
+                    RuntimeErrorCode::NoRouteFound,
+                    "Failed to find any route",
+                )),
+                RetryableSendFailure::DuplicatePayment => Err(runtime_error(
+                    RuntimeErrorCode::AlreadyUsedInvoice,
+                    "Already used invoice",
+                )),
```

Comment:
Our current payment tracking mechanism should prevent this case from happening. Maybe return a permanent error instead?